### ISSUE TITLE
Add lint for redundant parameter values

### ISSUE THREAD

**Issue by kenzieschmoll** [dart-lang/linter#1790]
`redundant_parameter_value`: warns when a user explicitly passes the default value for a parameter. 

```
class Foo {
  Foo({bar = ‘baz’});
}
var foo = Foo(bar: ‘baz’); // this should trigger the warning
```

**Comment by bwilkerson** [comment]
I suspect that this would be useful, but I always like to consider false positives. The one case I can think of is if the client wants to guard against the possibility that the default value might change by making the value explicit. I don't know how often that case would arise.

The other issue with a changed default value is that code that used to be clean might start generating a lint. Also probably not a frequent occurrence.

**Comment by jacob314** [comment]
This seems like a good lint to test out by evaluating against the existing corpus of dart packages. We should be able to quickly spot check a random sample of ~50 lint warnings to see how many look like they could plausibly be false positives. I think it is pretty easy in practice to tell the difference between accidentally using the default value and intentionally doing something different. We can reach out to package authors if any of the cases are ambiguous.

**Comment by pq** [comment]
> The one case I can think of is if the client wants to guard against the possibility that the default value might change by making the value explicit.

To this point, as @bwilkerson and I were just discussing, this is a great candidate for platform API migration tooling to help with.
